File: OpenDiablo2.Common/Models/SoundEntry.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OpenDiablo2.Common.Models
{
    public struct SoundEntry
    {
        public string Handle { get; set; }
        public int Index { get; set; }
        public string FileName { get; set; }
        public byte Volume { get; set; }
        public int GroupSize { get; set; }
        public bool Loop { get; set; }
        public int FadeIn { get; set; }
        public int FadeOut { get; set; }
        public int DeferInst { get; set; }
        public int StopInst { get; set; }
        public int Duration { get; set; }
        public int Compound { get; set; }
        public bool Reverb { get; set; }
        public int Falloff { get; set; }
        public int Cache { get; set; }
        public bool AsyncOnly { get; set; }
        public int Priority { get; set; }
        public int Stream { get; set; }
        public int Stereo { get; set; }
        public int Tracking { get; set; }
        public int Solo { get; set; }
        public int MusicVol { get; set; }
        public int Block1 { get; set; }
        public int Block2 { get; set; }
        public int Block3 { get; set; }
    }

    public static class SoundEntryHelper
    {
        public static SoundEntry ToSoundEntry(this string source)
        {
            var props = source.Split('\t');
            return new SoundEntry
            {
                Handle = props[0],
                Index = Convert.ToInt32(props[1]),
                FileName = props[2],
                Volume = Convert.ToByte(props[3]),
                GroupSize = Convert.ToInt32(props[4]),
                Loop = Convert.ToInt32(props[5]) == 1,
                FadeIn = Convert.ToInt32(props[6]),
                FadeOut = Convert.ToInt32(props[7]),
                DeferInst = Convert.ToInt32(props[8]),
                StopInst = Convert.ToInt32(props[9]),
                Duration = Convert.ToInt32(props[10]),
                Compound = Convert.ToInt32(props[11]),
                Reverb = Convert.ToInt32(props[12]) == 1,
                Falloff = Convert.ToInt32(props[13]),
                Cache = Convert.ToInt32(props[14]),
                AsyncOnly = Convert.ToInt32(props[15]) == 1,
                Priority = Convert.ToInt32(props[16]),
                Stream = Convert.ToInt32(props[17]),
                Stereo = Convert.ToInt32(props[18]),
                Tracking = Convert.ToInt32(props[19]),
                Solo = Convert.ToInt32(props[20]),
                MusicVol = Convert.ToInt32(props[21]),
                Block1 = Convert.ToInt32(props[22]),
                Block2 = Convert.ToInt32(props[23]),
                Block3 = Convert.ToInt32(props[24]),
            };
        }
    }
}

File: Sound/SoundEntry.go
package Sound

import (
	"github.com/essial/OpenDiablo2/Common"
	"strings"
)

// SoundEntry represents a sound entry
type SoundEntry struct {
	Handle    string
	Index     int
	FileName  string
	Volume    byte
	GroupSize uint8
	Loop      bool
	FadeIn    uint8
	FadeOut   uint8
	DeferInst uint8
	StopInst  uint8
	Duration  uint8
	Compound  int8
	Reverb    bool
	Falloff   uint8
	Cache     uint8
	AsyncOnly bool
	Priority  uint8
	Stream    uint8
	Stereo    uint8
	Tracking  uint8
	Solo      uint8
	MusicVol  uint8
	Block1    int
	Block2    int
	Block3    int
}

// CreateSoundEntry creates a sound entry based on a sound row on sounds.txt
func CreateSoundEntry(soundLine string) SoundEntry {
	props := strings.Split(soundLine, "\t")
	result := SoundEntry{
		Handle:    props[0],
		Index:     Common.StringToInt(props[1]),
		FileName:  props[2],
		Volume:    Common.StringToUint8(props[3]),
		GroupSize: Common.StringToUint8(props[4]),
		Loop:      Common.StringToUint8(props[5]) == 1,
		FadeIn:    Common.StringToUint8(props[6]),
		FadeOut:   Common.StringToUint8(props[7]),
		DeferInst: Common.StringToUint8(props[8]),
		StopInst:  Common.StringToUint8(props[9]),
		Duration:  Common.StringToUint8(props[10]),
		Compound:  Common.StringToInt8(props[11]),
		Reverb:    Common.StringToUint8(props[12]) == 1,
		Falloff:   Common.StringToUint8(props[13]),
		Cache:     Common.StringToUint8(props[14]),
		AsyncOnly: Common.StringToUint8(props[15]) == 1,
		Priority:  Common.StringToUint8(props[16]),
		Stream:    Common.StringToUint8(props[17]),
		Stereo:    Common.StringToUint8(props[18]),
		Tracking:  Common.StringToUint8(props[19]),
		Solo:      Common.StringToUint8(props[20]),
		MusicVol:  Common.StringToUint8(props[21]),
		Block1:    Common.StringToInt(props[22]),
		Block2:    Common.StringToInt(props[23]),
		Block3:    Common.StringToInt(props[24]),
	}
	return result
}
